### PR TITLE
CYBRSOURCE-55

### PR DESCRIPTION
**Pull request by btalma**
Ecuador payload customizations: add shipping tax as a line item in the format:
{
    "productName": "ADMINISTRACION MANEJO DE PRODUCTO", //same name always
    "productSKU": "", //not required by CS, can be omitted (preferable) or empty
    "quantity": "1", //always 1
    "unitPrice": "25.00", // Shipping cost without taxes
    "taxAmount": "25.00", // unitPrice * quantity
     "taxDetails": [
      {
        "type": "national",
        "amount": "3.00"  // taxAmount * taxRate (based on configuration in the account)
      }
    ]
  }

### CODE REVIEW

**Review by AnnaChiu95 — COMMENTED** [review]
Is `taxAmount` the taxable value? 

How do we test these changes?

**Comment by btalma** [comment]
"taxAmount": should be unitPrice * quantity
The change can be tested on sandboxusdev by making sure that the Region is set to Ecuador on the Cybersource admin page and checking out.  The payload can be viewed in OpenSource.
https://developer.logs.vtex.com/_dashboards/app/discover#/?_g=(filters:!(),refreshInterval:(pause:!t,value:0),time:(from:now-2h,to:now))&_a=(columns:!(_source),filters:!(('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'2cab9aa0-5c56-11ed-862d-710ea72622fc',key:Attributes.app,negate:!f,params:(query:'vtex.cybersource*'),type:phrase),query:(match_phrase:(Attributes.app:'vtex.cybersource*'))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'2cab9aa0-5c56-11ed-862d-710ea72622fc',key:Attributes.workflow_type,negate:!f,params:(query:CreatePayment),type:phrase),query:(match_phrase:(Attributes.workflow_type:CreatePayment))),('$state':(store:appState),meta:(alias:!n,disabled:!f,index:'2cab9aa0-5c56-11ed-862d-710ea72622fc',key:Attributes.account,negate:!f,params:(query:sandboxusdev),type:phrase),query:(match_phrase:(Attributes.account:sandboxusdev)))),index:'2cab9aa0-5c56-11ed-862d-710ea72622fc',interval:auto,query:(language:kuery,query:''),sort:!())

File: dotnet/Services/CybersourcePaymentService.cs [hunk 1]
```diff
@@ -565,15 +566,29 @@ public async Task<Payments> BuildPayment(CreatePaymentRequest createPaymentReque
...
-                            if ((name.Contains("tax@") || name.Contains("taxhub@")) && !name.Contains("shipping"))
+                            if (name.Contains("tax@") || name.Contains("taxhub@"))
                             {
-                                if (priceTag.IsPercentual ?? false)
+                                if (name.Contains("shipping"))
                                 {
-                                    itemTax += (long)Math.Round(vtexOrderItem.SellingPrice * priceTag.RawValue, MidpointRounding.AwayFromZero);
+                                    if (priceTag.IsPercentual ?? false)
+                                    {
+                                        shippingTax += (long)Math.Round(vtexOrderItem.SellingPrice * priceTag.RawValue, MidpointRounding.AwayFromZero);
```
Comment (polishq) [inline dotnet/Services/CybersourcePaymentService.cs:575]: `vtexOrderItem.SellingPrice` is incorrect ... the tax rate needs to be applied to the shipping cost, not the item cost. Maybe you can just store the shipping tax rate here, and then use it to calculate the tax when you build the shipping tax line item?

**Review by polishq — CHANGES_REQUESTED** [review]

**Review by polishq — APPROVED** [review]
LGTM

**Pull request merged**